service/src/main/java/uk/gov/hmcts/reform/fpl/service/MessageJudgeService.java

```diff
@@ -37,4 +52,31 @@ protected String buildMessageHistory(String message, String history, String send
 
         return join("\n \n", history, formattedLatestMessage);
     }
+
+    protected String getEmailAddressByRoleType(JudicialMessageRoleType roleType) {
+        if (JudicialMessageRoleType.JUDICIARY.equals(roleType)) {
+            return userService.getUserEmail();
+        } else if (JudicialMessageRoleType.CTSC.equals(roleType)) {
+            return ctscEmailLookupConfiguration.getEmail();
+        }
+
+        return EMPTY;
+    }
+
+    protected String resolveSenderEmailAddress(JudicialMessageRoleType roleType, String defaultEmail) {
+        String resolvedAddress = getEmailAddressByRoleType(resolveSenderRoleType(roleType));
+        return isNotEmpty(resolvedAddress) ? resolvedAddress : defaultEmail;
+    }
+
+    protected String resolveRecipientEmailAddress(JudicialMessageRoleType roleType, String defaultEmail) {
+        return (JudicialMessageRoleType.CTSC.equals(roleType))
+            ? getEmailAddressByRoleType(JudicialMessageRoleType.CTSC) : defaultEmail;
+    }
+
+    protected JudicialMessageRoleType resolveSenderRoleType(JudicialMessageRoleType defaultRoleType) {
+        if (isJudiciary()) {
```

Comment:
Not sure why we need this method, is JUDICIARY not in judicialMessageReply where it gets used?

Reply:
The senderRoleType drop down list doesn’t include Judiciary, as we don't want someone else to mis-claim themselves as Judiciary. 

Hence, the judicialMessageReply.senderType will remain empty if a judge is sending the message and I have to resolve the role type before storing the new message in about-to-submit stage